lms/djangoapps/course_api/api.py

```diff
@@ -0,0 +1,55 @@
+"""
+Course API
+"""
+
+from django.contrib.auth.models import User
+from rest_framework.exceptions import PermissionDenied
+
+from lms.djangoapps.courseware.courses import get_courses, get_course_with_access
+
+from .permissions import can_view_courses_for_username
+from .serializers import CourseSerializer
+
+
+def get_effective_user(requesting_user, target_username):
+    """
+    Get the user we want to view information on behalf of.
+    """
+    if target_username == requesting_user.username or None:
+        return requesting_user
+    elif can_view_courses_for_username(requesting_user, target_username):
+        return User.objects.get(username=target_username)
+    else:
+        raise PermissionDenied()
+
+
+def course_detail(requesting_user, username, course_key, request):
+    """
+    Return a single course identified by `course_key`
+
+    Arguments:
+        `requesting_user`: (User)
+
+        `username`: (string)
```

Comment:
Should you rename this to `target_username` to be consistent throughout this file?